build.assets/build-package.sh

```diff
@@ -125,9 +125,9 @@ else
 
     # set docker image appropriately
     if [[ "${PACKAGE_TYPE}" == "deb" ]]; then
-        DOCKER_IMAGE="public.ecr.aws/gravitational/fpm:debian8"
+        DOCKER_IMAGE="public.ecr.aws/gravitational/fpm:debian12-1.15.1-1"
```

Comment:
Why is this odd? I always prefer to pin exact versions so you know what you're getting.

I thought the ECR registry was immutable so I didn't generate any floating labels for the fpm image. I have since learnt that the public ECR registry is mutable so I could go back and add a floating label, but have not yet done so. At the moment this is the only label that will work.

What is the concern with using exact versions? This doesn't need any bots to auto-update as whoever revs the fpm version will be doing so for a reason and can bump the version here to use that.

Reply:
I'd just move the version to the top and keep all versions in the same place next to each other. It will be easier to update them.
IMO pinning the version is a good practice as it removes one additional variable from each build. The main downside is that we will need to remember to update it from time to time to not be stuck on the same exact version forever.

Reply:
> The main downside is that we will need to remember to update it from time to time to not be stuck on the same exact version forever.

Exactly, which is what I was trying to avoid. Can we add this to renovate to monitor and keep updated?

Reply:
I think one of my points got buried at the end. This is not a third-party image that we need to watch for updates. It is a first party image created from gravitational/docker-fpm. If anyone changes that, they are doing so to see those changes reflected in `build-package.sh`, so they are going to change the version there themselves. I guess someone could fix a security issue in `gravitationa/docker-fpm` and not know where it is used, so renovate might possible help there, but that's pretty narrow. It's also pretty easy for someone to search our github org for the image name, so it is not hard to find.

I don't know how we set up renovate for this. If someone can do that, I'd be happy to review it, but I don't want to block this PR on this. The reality is we just do not change this image - we've used the same one for the last 2 years and I've only changed it because i need an arm64 version of it so I need to rebuild it and that necessitated a version bump.

Can I proceed?

Reply:
Yup, it's far from the only place we've done this. Not an immediate blocker. Just something to note that we likely want to address anytime we pin something. Pinning + lack of automation to update said pins == sad times. :-)